payload/reggae/backend/make.d

```diff
@@ -87,11 +87,15 @@ struct Makefile {
             // add a dependency on the Makefile to reggae itself and the build description,
             // but only if not exporting a build
             auto srcDirs = _srcDirs.sort.uniq;
+            const flattenedInputs = chain(options.reggaeFileDependencies, srcDirs).join(" ");
             const rerunLine = "\t" ~ options.rerunArgs.join(" ") ~ "\n";
 
-            ret ~= fileName() ~ ": " ~ chain(options.reggaeFileDependencies, srcDirs.save).join(" ") ~ "\n";
+            ret ~= fileName() ~ ": " ~ flattenedInputs ~ "\n";
             ret ~= rerunLine;
-            ret ~= "\n" ~ srcDirs.save.map!(d => d ~ ":" ~ "\n" ~ rerunLine).join("\n");
+            // Add a dummy target for all Makefile inputs, with an empty recipe, just so
+            // that re-generating the Makefile isn't blocked by removed/renamed inputs.
+            // See ninja.d for more details.
+            ret ~= "\n" ~ flattenedInputs ~ ": ;\n";
```

Comment:
I've tested this with reggae's own `reggaefile.d`, and removing `.reggae/buildgen_main.d`. Doing it this way with a single target and an empty recipe leads to a single reggae re-run, not 2 re-runs as with the previous way (could potentially be even much more, depending on how many inputs were removed).